vuln/core/README.txt

```diff
@@ -0,0 +1,631 @@
+including restricting the acceptable characters.
+
+Introduce new `--security-revert={cvenum}` command line flag for selective
+reversion of specific CVE fixes allow the fix for CVE-2016-2216 to be
+selectively reverted using `--security-revert=CVE-2016-2216`
+
+--
+
+vulnerable: 4.x, 5.x
+patched: ^5.11.1, ^4.4.4
+description: buffer safeguard against accidental kNoZeroFill
+overview:
+To reproduce: try { Buffer(1e10); } catch (e) {} new Uint8Array(100);.
+
+To be affected, one would need to:
+
+Have any way how the user could make the API pass huge numbers to
+Buffer/SlowBuffer/Buffer.allocUnsafe, e.g. by sending invalid input
+
+--
+
+vulnerable: 4.x, 5.x
+patched: ^4.3.2, ^5.7.1
+ref: https://nodejs.org/en/blog/vulnerability/openssl-march-2016/
+ref: https://github.com/nodejs/node/pull/5507
+overview:
+Fix a double-free defect in parsing malformed DSA keys that may potentially be
+used for DoS or memory corruption attacks. It is likely to be very difficult to
+use this defect for a practical attack and is therefore considered low severity
+for Node.js users. More info is available at
+cve: CVE-2016-0705
+
+--
+
+vulnerable: 4.x, 5.x
+patched: ^4.3.2, ^5.7.1
+ref: https://nodejs.org/en/blog/vulnerability/openssl-march-2016/
+ref: https://github.com/nodejs/node/pull/5507
+cve: CVE-2016-0797
+overview:
+Fix a defect that can cause memory corruption in certain very rare cases
+relating to the internal `BN_hex2bn()` and `BN_dec2bn()` functions. It is
+believed that Node.js is not invoking the code paths that use these functions so
+practical attacks via Node.js using this defect are _unlikely_ to be possible.
+
+--
+
+vulnerable: 4.x, 5.x
+patched: ^4.3.2, ^5.7.1
+ref: https://nodejs.org/en/blog/vulnerability/openssl-march-2016/
+ref: https://github.com/nodejs/node/pull/5507
+ref: https://ssrg.nicta.com.au/projects/TS/cachebleed
+cve: CVE-2016-0702
+overview:
+Fix a defect that makes the _[CacheBleed
+Attack](https://ssrg.nicta.com.au/projects/TS/cachebleed/)_ possible. This
+defect enables attackers to execute side-channel attacks leading to the
+potential recovery of entire RSA private keys. It only affects the Intel Sandy
+Bridge (and possibly older) microarchitecture when using hyper-threading. Newer
+microarchitectures, including Haswell, are unaffected.
+
+--
+
+vulnerable: 4.x, 5.x
+patched: ^5.6.0, ^4.3.0
+description: mitigate against the Logjam attack
+overview:
+To mitigate against the Logjam attack, TLS clients now reject Diffie-Hellman
+handshakes with parameters shorter than 1024-bits, up from the previous limit of
+768-bits.
+
+--
+
+vulnerable: 5.x, 4.x
+patched: ^5.1.1, ^4.2.3
+cve: CVE-2015-8027
+overview:
+A bug whereby an HTTP socket may no longer have a parser associated with it but
+a pipelined request attempts to trigger a pause or resume on the non-existent
+parser, a potential denial-of-service vulnerability.
+
+--
+
+XXX this doesn't appear to be remote, is it a node vulnerability?
```

Comment:
OK, looks like its CVSS score indicated it was a vuln, and its possible to imagine data being injected which when stringified and returned would leak memory?